ludwig/benchmarking/reporting.py

```diff
@@ -141,49 +142,54 @@ def get_resource_usage_report(
     return info
 
 
-def get_all_events(
+def get_all_events(tags: set,
     kineto_events: List[_KinetoEvent], function_events: profiler_util.EventList
 ) -> Tuple[List[_KinetoEvent], List[profiler_util.FunctionEvent], List[Any], List[_KinetoEvent]]:
     """Return main Kineto and function events (tagged with "ludwig.*"), memory and out of memory events.
 
+    :param tags: set of code block/function tags we report resource usage metrics for.
     :param kineto_events: list of Kineto Events.
     :param function_events: list of function events.
     """
-    main_function_events = [evt for evt in function_events if "ludwig" in evt.name]
-    main_kineto_events = [event for event in kineto_events if "ludwig" in event.name()]
-    memory_events = [[event, False] for event in kineto_events if "[memory]" in event.name()]
+    main_function_events = [evt for evt in function_events if evt.name in tags]
+    main_kineto_events = [event for event in kineto_events if event.name() in tags]
+    memory_events = [[event, False] for event in kineto_events if profiler_util.MEMORY_EVENT_NAME in event.name()]
+    # profiler_util.OUT_OF_MEMORY_EVENT_NAME seems to only be in newer versions of torch.
     out_of_memory_events = [event for event in kineto_events if "[OutOfMemory]" in event.name()]
     return main_kineto_events, main_function_events, memory_events, out_of_memory_events
 
 
-def export_metrics_from_torch_profiler(profile: torch.profiler.profiler.profile, output_dir: str):
+def export_metrics_from_torch_profiler(tags: list, profile: torch.profiler.profiler.profile, output_dir: str):
```

Comment:
`tags` here is list, whereas in other places it is `set`, let's be consistent across. Additionally, can we specify what's the type of the items in the set/list? Is it strings/callables?